backfiller/backfiller/main.py

```diff
@@ -0,0 +1,200 @@
+
+# this is a prototype of the backfiller
+# lots about web services and the like I don't know
+# needs logging, exception handling and the like
+# also proper doc strings
+
+# when starting the backfiller and every few hours, backfill everything
+# more frequently, backfill the last couple hours
+# (last three hour directories so always at least two hours).
+
+import os
+import time
+import datetime
+import errno
+import uuid
+
+import requests
+
+import common
+
+HOUR_FMT = '%Y-%m-%dT%H'
+TIMEOUT = 5
+
+def get_nodes():
+
+	# either read a config file or query the database to get the addresses
+	# of the other nodes
+	# figure out some way that the local machine isn't in the list of returned
+	# nodes so that 
+
+	# as a prototype can just hardcode some addresses.
+	# each element in nodes is a 'protocol://host:port/' string
+
+	nodes = []
+
+	return nodes
+
+def list_local_segments(base_dir, stream, variant, hour):
+
+	# based on restreamer.list_segments
+	# could just call restreamer.list_segments but this avoids http/json
+	# overheads
+	path = os.path.join(base_dir, stream, variant, hour)
+	try:
+		local_segments = [name for name in os.listdir(path) if not name.startswith('.')]
```

Comment:
just directly return the value instead of assigning it to a variable to be returned later